src/deferred.js

```diff
@@ -212,20 +215,26 @@ jQuery.extend( {
 										}
 									};
 
+							processQueue.push( process );
+
 							// Support: Promises/A+ section 2.3.3.3.1
 							// https://promisesaplus.com/#point-57
 							// Re-resolve promises immediately to dodge false rejection from
 							// subsequent errors
 							if ( depth ) {
-								process();
+								deferredTick();
```

Comment:
No, because then things would probably fall out of spec. I'm instead advocating for broader changes that would eliminate the need for doing something different when `depth > 0`, at which point there won't be any queue jumping.